### PR TITLE
Refactor: Move TextEditor handling in its own component

### PR DESCRIPTION
**Pull request by mejo-**
Preparations for using the new editor API from Text, see https://github.com/nextcloud/text/pull/3615

Signed-off-by: Jonas <jonas@freesources.org>

### CODE REVIEW

File: src/components/Page/TextEditor.vue [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+	watch: {
+		'editMode'(val) {
+			if (val === true) {
+				this.startEdit()
+			} else {
+				this.stopEdit()
+			}
+		},
+
+		/*
```
Comment (juliushaertl) [inline src/components/Page/TextEditor.vue:95]: Still relevant?

**Review by juliushaertl — APPROVED** [review]
Change looks good. I added two follow up requirements from reading the diff to the text pr checklist:

- Things that collectives triggers
  - Save through syncService.save()
  - focusEditor

File: src/components/Page/TextEditor.vue (hunk 1 above)
Comment (mejo-) [reply src/components/Page/TextEditor.vue]: Thanks, removed now :blush:

**Review by max-nextcloud — APPROVED** [review]
Tested to confirm ui issue when pressing enter quickly after creating a new page is fixed.

**Pull request merged**